console/src/main/scripts/plugins/metrics/ts/directives/explorer/sideFilterContainerDirective.ts

```diff
@@ -0,0 +1,170 @@
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+
+/// <reference path="../../metricsPlugin.ts"/>
+/// <reference path="../../../includes.ts"/>
+
+module HawkularMetrics {
+
+  export class SideFilterController {
+    public selectedFeed: any;
+    public selectedResource: any;
+    public feedsConfig: any;
+    public resourceConfig: any;
+    public childrenTree: any[] = [];
+    /*@ngInject*/
+    constructor(private HawkularInventory: any) {
+      this.feedsConfig = {
+        title: 'Feed'
+      };
+
+      this.resourceConfig = {
+        title: 'Resource'
+      };
+      this.getFeeds();
+    }
+
+    private getFeeds() {
+      this.HawkularInventory.Feed.query({
+      }).$promise.then( (feedList) => {
+        this.feedsConfig.values = feedList;
+      });
+    }
+
+    public feedSelected(feed) {
+      this.HawkularInventory.ResourceUnderFeed.query({
+        feedId: feed.id
+      }).$promise.then((resourceList) => {
+        this.resourceConfig.values = resourceList;
+        this.selectedFeed = feed;
+      });
+    }
+
+    public resourceSelected(resource) {
+      this.selectedResource = resource;
+      if (this.hasOwnProperty('loadMetrics') && typeof this['loadMetrics'] === 'function') {
+        this['loadMetrics']({resource: this.selectedResource});
+      }
+      this.fetchResourceChildren(resource, this.selectedFeed);
+    }
+
+    private fetchResourceChildren(resource, feed) {
+      this.HawkularInventory.ResourceRecursiveChildrenUnderFeed.get({
+        feedId: feed.id,
+        resourcePath: resource.id
+      }).$promise.then((children) => {
+        let preffix = feed.id + ExplorerController.pathDelimiter;
+        let grouppedChildren = _.groupBy(children, (item: any) => item.path.substring(
+          item.path.lastIndexOf(preffix) + preffix.length,
+          item.path.lastIndexOf(ExplorerController.pathDelimiter))
+        );
+        const orderedChildren = {};
+        Object.keys(grouppedChildren).sort().forEach(function(key) {
+          orderedChildren[key] = grouppedChildren[key];
+        });
+        let childrenAsTree = _.reduce(orderedChildren, (curr: any, value: any, key: string) => {
+          const path = key.split(ExplorerController.pathDelimiter);
+          if (path.length === 1) {
+            curr.text = SideFilterController.stripSpecialCharsFromLabel(path[0]);
+            curr.nodes = SideFilterController.fillChildrenNodes(value);
+          } else {
+            path.shift();
+            this.insertAtPath(curr, path, value);
+          }
+          return curr;
+        }, _.cloneDeep(this.selectedResource));
+
+        if (children.length !== 0) {
+          this.childrenTree = [childrenAsTree];
+        } else {
+          this.childrenTree.length = 0;
+        }
+      });
+    }
+
+    private static stripSpecialCharsFromLabel(childId): string {
+      let label = childId;
```

Comment:
Not neccessary to assign `label = childId` because next line assigns it to something else.